メッセージ送信機能の非同期通信

hidetoshi1230 opened the pull request:
WHAT
メッセージ送信機能の非同期通信を実装

WHY
ユーザビリティ向上のため

hidetoshi1230 committed:
メッセージ送信機能の非同期通信１（実装）

jsファイルを作成する
フォームが送信されたら、イベントが発火するようにする
2のイベントが発火したときにAjaxを使用して、messages#createが動くようにする
messages#createでメッセージを保存し、respond_toを使用してHTMLとJSONの場合で処理を分ける
jbuilderを使用して、作成したメッセージをJSON形式で返す
返ってきたJSONをdoneメソッドで受取り、HTMLを作成する
6で作成したHTMLをメッセージ画面の一番下に追加する
HTMLを追加した分、メッセージ画面を下にスクロールする
非同期に失敗した場合の処理も準備する
※ターボリンク無効化

fumihumi commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,36 @@
+                      = ${ @message.body }`
+    return html;
+  }
+  $('#new_message').on('submit', function(e){
```
画像の連投って可能ですか？
確認お願いします。

fumihumi commented on app/views/messages/create.json.jbuilder:
```diff
@@ -0,0 +1,4 @@
+json.name @message.user.name
+json.date @message.created_at
```
message送信後のcreated_atの値ですが、加工しないと、見本chatspaceと表示が異なっていませんか？

fumihumi commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,36 @@
+                  .message--date
+                    = ${ @message.created_at }
+                  .message--text
+                    - if message.image?
```
画像とテキストって一緒に投稿できない仕様でしたっけ？

hidetoshi1230 committed:
メッセージ送信機能の非同期通信２（修正、スクロール機能）

・画像とテキストって一緒に投稿できない仕様でしたっけ？
・画像の連投って可能ですか？確認お願いします。
・message送信後のcreated_atの値ですが、加工しないと、見本chatspaceと表示が異なっていませんか？
＋
投稿時のスクロール機能追加

hidetoshi1230 commented:
【テキスト投稿】
https://gyazo.com/446cc61cea579934af96f873b9a9675d
【画像投稿】
https://gyazo.com/d8aae6fa4fbdfcb9a7dfbfddaf7abbe9

kouuuki commented:
LGTMです！
マージしてください！
